sabnzbd/articlecache.py

```diff
@@ -93,9 +93,14 @@ def save_article(self, article: Article, data: bytes):
         nzo.add_saved_article(article)
 
         if article.lowest_partnum and not article.nzf.import_finished:
-            # Write the first-fetched articles to disk
-            # Otherwise the cache could overflow
-            self.__flush_article_to_disk(article, data)
+            # If the filename is verified and there is room left then the first part
+            # will be saved to the correct file immediately, so it's kept in cache.
+            # Otherwise it will be saved temporarily to save cache space.
+            if article.nzf.filename_checked and self.space_left():
+                self.reserve_space(len(data))
```

Comment:
Shouldn't this just skip to the code below to follow the regular path? Otherwise we could also exceed cache limits.

Reply:
But only slightly. I commented on why I did it like that in the PR description. If you still disagree I will change it.

Reply:
I think it's wise to chose the safe route. Especially since it's just a minor optimization. It also prevents adding another code-path in case of problems, it just goes the normal path.